includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -690,6 +690,10 @@ public function process_payment_for_order( $cart, $payment_information ) {
 					$token->get_last4()
 				);
 				$order->add_order_note( $note );
+			} elseif ( $payment_information->is_using_saved_payment_method() ) {
+				// We need to make sure the saved payment method is saved to the order so we can
+				// charge the payment method for a future payment.
+				$this->add_token_to_order( $order, $payment_information->get_payment_token() );
```

Comment:
yeah, that looks like a good change, so long as we're not pulling it outside the

```php
if ( ! $payment_needed && ! $save_payment_method ) {
```

if statement 🤔 . I don't want to put it outside that if-statement, because it looks like we only want to save the token to the order in some specific circumstances.

Made the change in 4d85c76af90934df02a583062bf3b179799bb4d9

Reply:
> I don't want to put it outside that if-statement, because it looks like we only want to save the token to the order in some specific circumstances.

Agreed, if the token needs to be saved in other circumstances it'll be taken care of a few lines below.

> Made the change in 4d85c76

🔥